server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -437,13 +437,38 @@ protected final void closeNoLock(String reason, CountDownLatch closedLatch) {
                     latestSegmentInfos.counter = latestSegmentInfos.counter + SI_COUNTER_INCREMENT;
                     latestSegmentInfos.changed();
                 }
-                commitSegmentInfos(latestSegmentInfos);
-                IOUtils.close(readerManager, translogManager, store::decRef);
+                try {
+                    commitSegmentInfos(latestSegmentInfos);
+                } catch (IOException e) {
+                    // mark the store corrupted unless we are closing as result of engine failure.
+                    // in this case Engine#failShard will handle store corruption.
+                    if (failEngineLock.isHeldByCurrentThread() == false && store.isMarkedCorrupted() == false) {
+                        try {
+                            store.markStoreCorrupted(e);
+                        } catch (IOException ex) {
+                            logger.warn("Unable to mark store corrupted", ex);
+                        }
+                    }
+                }
+                try {
+                    IOUtils.close(readerManager);
+                } catch (Exception e) {
+                    logger.warn("Failed to close reader manager");
+                }
+                try {
+                    IOUtils.close(translogManager);
+                } catch (Exception e) {
+                    logger.warn("Failed to close translog");
+                }
```

Comment:
You're swallowing the exceptions here without logging the details. Are you sure this is preferred over `IOUtils.close(readerManager, translogManager)` which will accumulate any exceptions and they'll get logged in the catch block?

Reply:
oops good catch.  You're right will leave as it was, as long as we move store decref to finally block it solves the issue.